fb-negotiation/core/split_tracker.py

```diff
@@ -0,0 +1,327 @@
+          mention['agent'] = 'for_me'
+          agent_found = True
+          break
+      if agent_found:
+        continue
+      for token in mention['after']:
+        if token == "me":
+          mention['agent'] = 'for_them'
+          agent_found = True
+          break
+        elif token in ["you"]:
+          mention['agent'] = 'for_me'
+          agent_found = True
+          break
+      if agent_found:
+        continue
+      # by default, if nothing matches, we assume the offer was for them
+      mention['agent'] = 'for_them'
+
+  def merge_their_offers(self):
+    for item in self.items:
+      if self.made['their_offer_for_me'] == True:
+        # if they already stated what they want, then do not mess with it
+        if self.their_offer[item] >= 0:
+          for_them = self.their_offer[item]
+        # they should get the opposite of whatever they believe I should have
+        elif self.their_offer[item] < 0 and self.their_offer_for_me[item] >= 0:
+          for_them = self.item_counts[item] - self.their_offer_for_me[item]
+        # if they proposed something, assume anything not mentioned is for them
+        else:
+          for_them = self.item_counts[item]
+        self.their_offer[item] = for_them
+      elif self.their_offer[item] < 0:
+        self.their_offer[item] = 0
+
+  def unit_test(self, c, d, raw_utterance):
+    scenario = {'book':c[0] , 'hat':c[1], 'ball':c[2]}
+    self.set_item_counts(scenario)
+
+    self.build_lexicon(tokenize(raw_utterance))
+    self.determine_item_count()
+    self.determine_which_agent()
+    self.resolve_tracker()
+    self.merge_their_offers()
+
+    book_match = self.their_offer['book'] == d[0]
+    hat_match = self.their_offer['hat'] == d[1]
+    ball_match = self.their_offer['ball'] == d[2]
+
+    if book_match and hat_match and ball_match:
+      print("Passed")
+      passed = True
+    else:
+      print("TEST SCENARIO")
+      print("  There are {0} books, {1} hats, and {2} balls.".format(c[0], c[1], c[2]) )
+      print("  Sentence: {0}".format(raw_utterance) )
+      print("SYSTEM OUTPUT")
+      print("  They want {0} books, {1} hats, and {2} balls".format(
+          self.their_offer['book'], self.their_offer['hat'], self.their_offer['ball']))
+      for item in self.items:
+        ct = self.their_offer_for_me[item]
+        if ct > 0:
+          print("  They think I should should get {0} {1}s".format(ct, item))
+      print("  The correct split is {0} books, {1} hats, and {2} balls".format(d[0], d[1], d[2]) )
+      print("LEXICON")
+      for mention in self.lexicon:
+        print mention
+      passed = False
+
+    print("------------------------------")
+    return passed
+
+  @staticmethod
+  def clarify():
+    print("Ask for clarification")
+    # If you heard no items:
+    #   check for question mark
+
+  def link_entity(self, raw_tokens, kb=None, scale=True, price_clip=None):
+    tokens = ['<s>'] + raw_tokens + ['</s>']
```

Comment:
As mentioned in the above comment, the current tracker is a bit different from “lexicons" for the other two tasks, which is fine. But then the `link_entity` function is not needed and this should not be in `core`.

Reply:
Totally, link_entity is legacy code, it's gone!